streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -288,7 +410,9 @@ public String toString() {
      * @throws IllegalStateException if instance is currently running
      */
     public void cleanUp() {
-        if (state == StreamsState.running) {
+        State tmpState = state();
+        if (tmpState == KafkaStreams.State.RUNNING ||
```

Comment:
you could encapsulate a lot of these small pieces of logic in to your enum, so instead have 
```
public enum State { 
CREATED, RUNNING, REBALANCING, PENDING_SHUTDOWN, NOT_RUNNING;
  public boolean canCleanup() {
    return !(this.equals(RUNNING) || this.equals(REBALANCING));
  }
}
```

would also allow you to get rid of the `tmpState` variables hanging around this file.